### PR TITLE
Graceul termination in Cacher

### PR DESCRIPTION
**Pull request by wojtek-t**
This also eliminates a lot of rubbish logs in unit tests.

### CODE REVIEW

File: pkg/storage/cacher.go [hunk 1]
```diff
@@ -150,14 +153,31 @@ func NewCacherFromConfig(config CacherConfig) *Cacher {
 		watchers:   make(map[int]*cacheWatcher),
 		versioner:  config.Versioner,
 		keyFunc:    config.KeyFunc,
+		stopped:    false,
+		// We make stopCh a buffered channel, because we potentially need to stop both:
+		// - util.Until go-routine
+		// - reflector.ListAndWatch
+		// but there are no guarantees to the order that we need to stop them.
```
Comment (gmarek) [inline pkg/storage/cacher.go]: ...there are no guarantees on the order that they will stop, so we make the channel...

Comment (wojtek-t) [inline pkg/storage/cacher.go]: done

**Comment by wojtek-t** [comment]
@gmarek - PTAL (I applied your comment we discussed offline)

**Comment by gmarek** [comment]
LGTM

**Pull request merged**